storage/aptosdb/src/lib.rs

```diff
@@ -986,6 +986,17 @@ impl DbReader for AptosDB {
         })
     }
 
+    fn get_first_block_height(&self) -> Result<Option<u64>> {
+        gauged_api("get_first_block_height", || {
+            if let Some(version) = self.get_first_txn_version()? {
+                let (_, _, event) = self.get_block_info(version)?;
```

Comment:
Man this faces the same issue: the new block event can be gone for `version` because of the pruner, and `self.get_block_info()` will return an error.

It's tricky to get this accurate and not prone to errors.

Also notice the pruner is continuously running, so as soon as you get the first block height, it's possible the block is already pruned.. what's the use case for this API?

Reply:
I just need some semblance of an oldest unpruned block so when I call `/block` I know it's not pruned

Reply:
In that case calling lookup_event_at_or_after_version() you added above instead of get_block_info() should do?

Reply:
Yes, exactly!  Should be good enough, it's still a group where it's continuous, and that's all I need